src/api/environment.cc

```diff
@@ -268,25 +291,42 @@ Local<Context> NewContext(Isolate* isolate,
                            True(isolate));
 
   {
-    // Run lib/internal/bootstrap/context.js
+    // Run per-context JS files.
     Context::Scope context_scope(context);
-
-    std::vector<Local<String>> parameters = {
-        FIXED_ONE_BYTE_STRING(isolate, "global")};
-    Local<Value> arguments[] = {context->Global()};
-    MaybeLocal<Function> maybe_fn =
-        per_process::native_module_loader.LookupAndCompile(
-            context, "internal/bootstrap/context", &parameters, nullptr);
-    if (maybe_fn.IsEmpty()) {
-      return Local<Context>();
-    }
-    Local<Function> fn = maybe_fn.ToLocalChecked();
-    MaybeLocal<Value> result =
-        fn->Call(context, Undefined(isolate), arraysize(arguments), arguments);
-    // Execution failed during context creation.
-    // TODO(joyeecheung): deprecate this signature and return a MaybeLocal.
-    if (result.IsEmpty()) {
+    Local<Object> exports;
+    if (!GetPerContextExports(context).ToLocal(&exports))
       return Local<Context>();
+
+    Local<String> global_string = FIXED_ONE_BYTE_STRING(isolate, "global");
+    Local<String> exports_string = FIXED_ONE_BYTE_STRING(isolate, "exports");
+
+    static const char* context_files[] = {
+      "internal/per_context/setup",
```

Comment:
It seems a bit strange that this file is named as a generic`setup`...if we move the `Intl.v8BreakIterator` deletion back to C++ then that file can be just something like `warn_atomics_await` (and can be deleted soon, I just saw the v8 change that renamed the thing landed today)

Also, do we need to support this for every context? Even for the ones that do not run through bootstrapping? What's the use case of moving the port to a regular vm context?

Reply:
I think it might be nice to keep a way to run a number of scripts for every Node.js context, so that we could extend some of our functionalities to generic vm Contexts.

> Even for the ones that do not run through bootstrapping?

@joyeecheung What do you mean? The `vm.createContext()` ones? That’s the target for this patch, yes. Moving to other contexts isn’t currently possible anyway?

> What's the use case of moving the port to a regular vm context?

It’s the better Isolate we get from vm Contexts, and it would make it easier to implement something like Web-style Workers; those have a custom global object with special properties, so using the `vm` module should make sense.